cmd/algocfg/profileCommand.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"path/filepath"
+	"strings"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/util/codecs"
+)
+
+var (
+	profileNames = []string{"relay", "default"}
+)
+
+func init() {
+	rootCmd.AddCommand(profileCmd)
+	profileCmd.AddCommand(setProfileCmd)
+	profileCmd.AddCommand(listProfileCmd)
+}
+
+var profileCmd = &cobra.Command{
+	Use:   "profile",
+	Short: "Manipulate config profiles",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.HelpFunc()(cmd, args)
+	},
+}
+
+var listProfileCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List config profiles",
+	Args:  cobra.NoArgs,
+	Run: func(cmd *cobra.Command, args []string) {
+		reportInfof("%v", strings.Join(profileNames, " "))
+	},
+}
+
+var setProfileCmd = &cobra.Command{
+	Use:   "set",
+	Short: "Set preconfigured config defaults",
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		onDataDirs(func(dataDir string) {
+			cfg, err := getConfigForArg(args[0])
+			if err != nil {
+				reportErrorf("%v", err)
+			}
+			file := filepath.Join(dataDir, config.ConfigFilename)
+			err = codecs.SaveNonDefaultValuesToFile(file, cfg, config.GetDefaultLocal(), nil, true)
+			if err != nil {
+				reportErrorf("Error saving updated config file '%s' - %s", file, err)
+			}
+		})
+	},
+}
+
+func getConfigForArg(configType string) (config.Local, error) {
+	switch configType {
+	case "relay":
+		return config.GetRelayLocal(), nil
+	case "default":
+		return config.GetDefaultLocal(), nil
+	default:
+		return config.Local{}, fmt.Errorf("invalid profile type %v", configType)
+	}
```

Comment:
Thinking out loud.

The configuration scheme for `config.GetDefaultLocal()` is really complicated, I wonder if we need to follow its pattern for this.

As an alternative, what do you think about treating profiles as overrides on the current default?
```suggestion
func getConfigForArg(configType string) (config.Local, error) {
	switch configType {
	case "relay":
		default := config.GetRelayLocal()
		default.Archival = true
		default.EnableLedgerService = true
		default.EnableBlockService  = false
		default.NetAddress = 4160
	case "default":
		return config.GetDefaultLocal(), nil
	default:
		return config.Local{}, fmt.Errorf("invalid profile type %v", configType)
	}
```

Reply:
Yeah, I updated to do that. I think what I wanted--and what will not be possible using this approach--is versioning and migration of defaults over time. It's hard to get that given how Local is currently setup, but I'm not sure the complexity of the original approach is worth it.